docs/docs/configuration/providers/ms_entra_id.md

```diff
@@ -0,0 +1,199 @@
+Azure has a limit of 200 groups in the JWT. If you can't avoid such a bug number and still want to access the groups, you need to grant `GroupMember.Read.All` delegated permission to the app registration so oauth2-proxy can read all the groups from Graph API. **NOTE**: This permission by default requires an admin consent!
+<details>
+    <summary>See Azure Portal example</summary>
+    <div class="videoBlock">
+        <video controls>
+            <source src="/oauth2-proxy/azure-videos/group-overage.mkv"/>
+        </video>
+    </div>
+</details>
+<details>
+    <summary>See Terraform example</summary>
+
+    resource "azuread_application" "auth" {
+        display_name     = "oauth2-proxy"
+        sign_in_audience = "AzureADMyOrg" # Other alre also supported
+
+        group_membership_claims = [
+            "SecurityGroup"
+        ]
+        optional_claims {
+            id_token {
+                name = "groups"
+            }
+        }
+
+        web {
+            redirect_uris = [
+                "https://podinfo.lakis.tech/oauth2/callback",
+            ]
+        }
+
+        required_resource_access {
+            resource_app_id = "00000003-0000-0000-c000-000000000000" # Microsoft Graph
+            resource_access {
+                id   = "e1fe6dd8-ba31-4d61-89e7-88639da4683d" # User.Read delegated
+                type = "Scope"
+            }
+            
+            resource_access {
+                id   = "bc024368-1153-4739-b217-4326f2e966d0" # GroupMember.Read.All delegated - admin consent required
+                type = "Scope"
+            }
+
+        }
+    }
+
+    resource "azuread_application_password" "apppass" {
+        application_id = azuread_application.auth.id
+    }
+
+Admin consent is required after creation by Terraform
+</details>
+
+## Configure provider
+Provider is OIDC-compliant, so all the OIDC parameters are honoured. Additional provider-specific configuration parameters are:
+* `ms-entra-id-skip-groups-from-graph` - never read groups from Graph API, even when ID token indicates that there's a group overage. Set if you expect group overage in some cases, but still don't want to assign wide `GroupMember.Read.All`. Defaults to `false`. If you don't need groups, consider skipping `groups` claim in the app registration.
+* `ms-entra-id-multi-tenant-allowed-tenant` - speciy list of allowed tenants to be authenticated through multi-tenant app. When not set, all tenants are allowed. Defaults to `[]` (all tenants).
+
+### Scope
+For Azure-only apps (multi-tenant and single-tenant), the only required oAuth scope is `openid`:
+```
+- --scope=openid
+```
+For personal MS accounts, the scope has to be extended with `email` and `profile`:
+```
+- --scope=openid profile email
+```
+It's recommended to configure the scopes explicitly, otherwise you may experience issues with allowing groups (Azure doesn't support `groups` scope which is automatically included when you configure allowed groups).
```

Comment:
```suggestion
It's recommended to configure the scopes explicitly, otherwise, you may experience issues with allowing groups (Azure doesn't support the `groups` scope, which is automatically included when you configure allowed groups).
```